### ISSUE TITLE
Currency select drop down does not auto-update

### ISSUE THREAD

**Issue by jameshilliard** [spree-contrib/spree_multi_currency#24]
gemfile.lock https://gist.github.com/jameshilliard/bcca2e8b9fc15e213c9d
steps to reproduce:
enable multi currency support
allow user to change currency
select alternate currency using dropdown menu
currency does not update when selecting new currency

**Comment by alepore** [comment]
yes, when a cart item is not available on the requested currency an exception is raised, like:
`RuntimeError (no RUB price found for Ruby on Rails Tote (ROR-00011)):`
and the submit will silently fail.

we need a strategy to handle this situation, like removing the product from the cart and/or display a proper message to the user.

**Comment by aishmita-aggarwal** [comment]
@priyank-gupta @damianlegawiec Currently after user updates currency, we are raising exception for line items that do not have price with selected currency. This tends to mislead the user.
I think we should show unavailability error for the line item and keep the currency unchanged.
What are your thoughts for this?

**Comment by priyank-gupta** [comment]
@aishmita-aggarwal Yeah, unavailability error is good than raising exceptions.
@damianlegawiec Thoughts?

**Comment by Mafi88** [comment]
```I think we should show unavailability error for the line item and keep the currency unchanged.```

@aishmita-aggarwal You mean something like ```If you want to change currency, you need to remove these products from cart: (list)```?

I think it's better than just removing products withproper  message.
@damianlegawiec What's your opinion?

**Comment by aishmita-aggarwal** [comment]
"I think it's better than just removing products withproper message."

@Mafi88 Those variants do not have prices with the updated currency. So are you suggesting that we should remove the unavailable items and just notify the user?

@priyank-gupta @damianlegawiec ?

**Comment by Mafi88** [comment]
@aishmita-aggarwal No, I think message like this: ```If you want to change currency, you need to remove these products from cart: (list)``` is better.
